test/replication-luatest/persistent_gc_anon_test.lua

```diff
@@ -227,3 +239,193 @@ g.test_fetch_snapshot_on_connected_replica = function(g)
+    local uuid = require('uuid').str()
+    g.server:exec(function() box.cfg{replication_anon_gc_timeout = 0.1} end)
+
+    g.server:exec(function()
+        -- Set the tweak to flush packets on each row and the tweak
+        -- not to send the read-view too fast
+        local tweaks = require('internal.tweaks')
+        tweaks.xrow_stream_flush_size = 1
+        box.error.injection.set('ERRINJ_RELAY_TIMEOUT', 0.2)
+    end)
+    write_fetch_snapshot(g.s, uuid)
+    local h = socket_read(g.s)
+    t.assert_equals(h[key.REQUEST_TYPE], type.OK)
+
+    fiber.sleep(0.3)
+
+    g.server:exec(function()
+        -- Consumer is still registered
+        t.assert_not_equals(box.space._gc_consumers:select{}, {})
+        t.assert_not_equals(box.info.gc().consumers, {})
+    end)
+
+    socket_restart(g)
+
+    -- Wait while the consumer will be deleted
+    g.server:exec(function()
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.space._gc_consumers:select{}, {})
+        end)
+    end)
+end
+
+g.test_expiration_subscribe = function(g)
+    local fiber = require('fiber')
+    g.server:exec(function()
+        box.cfg{replication_anon_gc_timeout = 0.1}
+    end)
+    local replicaset_uuid = g.server:eval("return box.info.replicaset.uuid")
+    local uuid = require('uuid').str()
+    local vclock = g.server:get_vclock()
+    vclock[0] = nil
+    write_subscribe(g.s, uuid, replicaset_uuid, vclock)
+    local h = socket_read(g.s)
+    t.assert_equals(h[key.REQUEST_TYPE], type.OK)
+
+    -- Sleep for a while and check if the consumer is still alive
+    fiber.sleep(0.2)
+    g.server:exec(function()
+        t.assert_equals(#box.space._gc_consumers:select{}, 1)
+        t.assert_equals(#box.info.gc().consumers, 1)
+    end)
+
+    socket_restart(g)
+
+    g.server:exec(function()
+        -- Wait while the consumer will be deleted
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.gc().consumers, {})
+            t.assert_equals(box.space._gc_consumers:select{}, {})
+        end)
+    end)
+
+    -- Expiration is logged
+    local msg = string.format(EXPIRATION_MSG_PATTERN, escape_uuid(uuid))
+    t.assert(g.server:grep_log(msg))
+end
+
+-- The case checks if the consumer is not expired when the replica is subscribed
+-- but receives nothing so its consumer is not advanced actually.
+g.test_expiration_idle_subscribe = function(g)
+    local fiber = require('fiber')
+    local uuid = require('uuid').str()
+    local replicaset_uuid = g.server:eval("return box.info.replicaset.uuid")
+    g.server:exec(function() box.cfg{replication_anon_gc_timeout = 0.2} end)
+    local vclock = g.server:get_vclock()
+    vclock[0] = nil
+    write_subscribe(g.s, uuid, replicaset_uuid, vclock)
+    local h = socket_read(g.s)
+    t.assert_equals(h[key.REQUEST_TYPE], type.OK)
+
+    -- Keep connected to master
+    for _ = 1, 10 do
+        fiber.sleep(0.1)
+        write_ok(g.s, {[key.VCLOCK] = setmap({})})
+    end
```

Comment:
Can we please make it a bit faster? Such a simple test takes more than a second now. Reduce the timeout like twice here, would it still cover what you need?

Reply:
Reduced all the timeouts (including `replication_anon_gc_timeout` and number of iterations.